java/java-impl-inspections/src/com/intellij/codeInspection/CastCanBeReplacedWithVariableInspection.java

```diff
@@ -130,10 +136,29 @@ private static boolean isChangedBetween(@NotNull final PsiVariable variable,
       controlFlow = ControlFlow.EMPTY;
     }
     int startOffset = controlFlow.getEndOffset(start) + 1;
-    int endOffset = controlFlow.getEndOffset(end);
+    int endOffset = controlFlow.getEndOffset(broadEnd);
```

Comment:
This test starts to fail after your changes, but was correctly working before:

```java
// "Replace '(String) obj' with 's'" "true"

class X {
  void test(Object obj) {
    for (int i = 0; i < 10; i++) {
      String s = (String) obj;
      System.out.println(s);
      System.out.println((String<caret>) obj);
      s = null;
    }
  }
}
```

I will commit this test. Please make sure that it works.

Reply:
Handled, and I have integrated the test

FYI, I tried to test the inspection but I failed to make the warning appear; I don't know why